components/script/dom/gpubuffer.rs

```diff
@@ -56,85 +56,22 @@ pub struct GPUBufferMapInfo {
     pub map_mode: Option<u32>,
 }
 
-#[dom_struct]
-pub struct GPUBuffer {
-    reflector_: Reflector,
-    #[ignore_malloc_size_of = "defined in webgpu"]
+#[derive(MallocSizeOf)]
+struct DroppableField {
+    cx: JSContext,
     channel: WebGPU,
-    label: DomRefCell<Option<USVString>>,
     state: Cell<GPUBufferState>,
     buffer: WebGPUBuffer,
     device: Dom<GPUDevice>,
-    size: GPUSize64,
     #[ignore_malloc_size_of = "promises are hard"]
     map_promise: DomRefCell<Option<Rc<Promise>>>,
```

Comment:
Yikes, interacting with the promise from the destructor is very scary, since it's a JS object. We'll need to think about what to do with this type.